packages-with-patches/download-rhel.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python3
+
+import click
+import re
+import requests
+from urllib.parse import urljoin
+from pathlib import Path
+
+CGIT = "https://pkgs.devel.redhat.com/cgit"
+# repositories starting with these prefixes are (probably) not RHEL packages
+IGNORE_PREFIXES = [
+    "Application-",
+    "Application_",
+    "CloudForms-",
+    "Engineering_",
+    "Fuse_",
+    "Hosted_",
+    "IEEE_",
+    "Internal-",
+    "JBoss_",
+    "OpenShift-",
+    "OpenShift_",
+    "Partner_",
+    "Policy-",
+    "Product_",
+    "RedHat_",
+    "Red_Hat_",
+    "Release_",
+    "Report-",
+]
+REPO_LINK = re.compile(
+    "<tr><td class='toplevel-repo'><a title='.+?' href='(?P<url>.+?)'>rpms/(?P<name>.+?)</a></td>"
+)
+SPECFILE = re.compile(
+    "<tr><td class='ls-mode'>.+?<a class='ls-blob spec' href='(?P<spec_url>.+?)'>"
+)
+
+
+def any_branch(url, branches):
+    """Check if a repo on a given URL has at least one of the branches
+
+    Args:
+        url: URL of the repo
+        branches: list of branches to be checked
+    Returns:
+        The name of the branch which is found the first or None if none of
```

Comment:
doesn't it return the last found branch?

Reply:
Right, I fixed the doc-string and also how to call this script, when there are multiple branches to pick from.

(Originally I started with going from the latest version to the newest, but then I realised it would result in fewer calls—and shorter execution time—if it's done the other way around.)